custom user model

nwolff opened the pull request:
In my project I use a custom user model that does not have a username field. See the official documentation for an example: https://docs.djangoproject.com/en/1.11/topics/auth/customizing/#django.contrib.auth.models.CustomUser

django-request chokes with such a model (both in the middleware and in the admin) with an unknown field error.

This pull-request replaces usage of the username field (which may not be present in the user model) by usage of the get_username() method that is always there.

This pull-request solves the same problem as https://github.com/django-request/django-request/pull/132 , but without requiring any extra configuration.

I tried to add extra tests to verify that django-requests works even with a custom user model (see trial here: https://github.com/skioo/django-request/commit/4f5529f7616e28704ed9ee3fbc37fb2a5f58d8da ) but when I swap out the default user model by setting AUTH_USER_MODEL to 'tests.MyUser' then  the test fails because the Request model was loaded at the beginning of the test and the user field expects a django.contrib.auth.models.User. Any help with that test would be appreciated.

felixxm commented on request/templates/request/plugins/activeusers.html:
```diff
@@ -9,7 +9,7 @@
     {% for user in user_list %}
         <tr>
-            <td><a href="{{ user.get_absolute_url }}">{{ user.username }}</a></td>
+            <td><a href="{{ user.get_absolute_url }}">{{ user.get_username() }}</a></td>
```
You can remove parentheses `{{ user.get_username }}`.

felixxm commented on request/templatetags/request_tag.py:
```diff
@@ -53,7 +53,7 @@ def active_users(parser, token):
         {% active_users in 10 minutes as user_list %}
         {% for user in user_list %}
-            {{ user.username }}
+            {{ user.get_username() }}
```
You can remove parentheses `{{ user.get_username }}`.

felixxm commented on request/admin.py:
```diff
@@ -39,7 +39,7 @@ def request_from(self, obj):
             return format_html(
                 '<a href="?user__username={0}" title="{1}">{2}</a>',
-                user.username,
+                user.get_username(),
```
I think it will be better to use a primary key (`user.pk`) in the filter instead of `get_username()`, i.e.:
```python
return '<a href="?user={0}" title="{1}">{2}</a>'.format(
    user.pk,
    _('Show only requests from this user.'),
    user,
)
```
I haven't noticed that before, sorry.

nwolff commented:
Thank you for catching that. I implemented your fix and also removed the lookup_allowed override for user__username that we don't use anymore. But now coveralls is unhappy because without those two lines the coverage went down a tiny bit

nwolff commented:
So I added a change to speedup the list of requests together with a very small test case.

coveralls is unhappy because there is one line less code now. Here is a discussion by a user having similar concerns: https://github.com/lemurheavy/coveralls-public/issues/565 , it also describes how to make coveralls a little less sensitive, if you are ready for that.

Cheers

felixxm commented:
Thanks.
